Watchman.Discord/Areas/Protection/RequestParsers/WarnRequestParser.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Devscord.DiscordFramework.Commons.Exceptions;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Watchman.DomainModel.Messages;
+using Watchman.DomainModel.Users;
+
+namespace Watchman.Discord.Areas.Protection.Services
+{
+    public class WarnRequestParser : RequestParser
+    {
+        public WarnRequestParser(DiscordRequest request, UsersService usersService, Contexts contexts) : base(request, usersService, contexts)
+        {
+        }
+
+        public WarnEvent GetWarnEvent()
+        {
+            var granterContext = _usersService.GetUserById(_contexts.Server, _contexts.User.Id);
+            var userToWarnContext = GetUser();
+
+            var grantor = new User(granterContext.Id, granterContext.Name);
+            var receiver = new User(userToWarnContext.Id, userToWarnContext.Name);
+
+            var reason = _request.Arguments.FirstOrDefault(x => x.Name == "reason" || x.Name == "r")?.Value;
+            var warnTime = _request.SentAt;
+            return new WarnEvent(grantor, receiver, reason, warnTime, _contexts.Server.Id);
+        }
+
+        public ulong GetServerIdForWarns()
+        {
+            bool allServers = _request.Arguments.Any(x => x.Name == "all");
```

Comment:
po pierwsze var
po drugie nazwa nie pasuje do typu bool 
po trzecie po co osobna zmienna, można to zapisać w ifie